plugins/modules/proxysql_query_rules_fast_routing.py

```diff
@@ -0,0 +1,470 @@
+            "AND schemaname = %s "
+            "AND flagIN = %s"
+        )
+
+        query_data = [
+            self.config_data["username"],
+            self.config_data["schemaname"],
+            self.config_data["flagIN"],
+        ]
+
+        cursor.execute(query_string, query_data)
+        check_count = cursor.fetchone()
+        return (int(check_count['rule_count']) > 0)
+
+    def check_rule_cfg_exists(self, cursor):
+        query_string = "SELECT count(*) AS `rule_count` FROM mysql_query_rules_fast_routing"
+
+        cols = 0
+        query_data = []
+
+        for col, val in iteritems(self.config_data):
+            if val is not None:
+                cols += 1
+                query_data.append(val)
+                if cols == 1:
+                    query_string += " WHERE " + col + " = %s"
+                else:
+                    query_string += " AND " + col + " = %s"
+
+        if cols > 0:
+            cursor.execute(query_string, query_data)
+        else:
+            cursor.execute(query_string)
+        check_count = cursor.fetchone()
+        return int(check_count['rule_count'])
+
+    def get_rule_config(self, cursor):
+        query_string = (
+            "SELECT * "
+            "FROM mysql_query_rules_fast_routing "
+            "WHERE username = %s "
+            "AND schemaname = %s "
+            "AND flagIN = %s"
+        )
+
+        query_data = [
+            self.config_data["username"],
+            self.config_data["schemaname"],
+            self.config_data["flagIN"]
+        ]
+
+        for col, val in iteritems(self.config_data):
+            if val is not None:
+                query_data.append(val)
+                query_string += "\n  AND " + col + " = %s"
+
+        cursor.execute(query_string, query_data)
+        rule = cursor.fetchall()
+        return rule
+
+    def create_rule_config(self, cursor):
+        query_string = \
+            """INSERT INTO mysql_query_rules_fast_routing ("""
+
+        cols = 0
+        query_data = []
+
+        for col, val in iteritems(self.config_data):
+            if val is not None:
+                cols += 1
+                query_data.append(val)
+                query_string += "\n" + col + ","
+
+        query_string = query_string[:-1]
+
+        query_string += (
+            ")\n" +
+            "VALUES (" +
+            "%s ," * cols
+        )
+
+        query_string = query_string[:-2]
+        query_string += ")"
+
+        cursor.execute(query_string, query_data)
+        return True
+
+    def update_rule_config(self, cursor):
+        query_string = "UPDATE mysql_query_rules_fast_routing"
+
+        cols = 0
+        query_data = [
+            self.config_data["username"],
+            self.config_data["schemaname"],
+            self.config_data["flagIN"]
+        ]
+
+        for col, val in iteritems(self.config_data):
+            if val is not None and col not in ("username", "schemaname", "flagIN"):
```

Comment:
```suggestion
            if val is not None and col not in ("username", "schemaname", "flagIN"):
```
if you also run this check when invoking the method in `main`, it would be good to get rid of one (i'd remove that one in `main` because it looks pretty long). If we have similar double checks somewhere in the code, IMO, would be nice to get rid of them as well.